benchmarks/distributed/rpc/ps/metrics/CUDAMetric.py

```diff
@@ -0,0 +1,25 @@
+import torch
+
+from .MetricBase import MetricBase
+
+
+class CUDAMetric(MetricBase):
+    def __init__(self, name: str):
+        self.name = name
+        self.start = None
+        self.end = None
+
+    def record_start(self, rank):
```

Comment:
Do we need to pass in rank here, or can it be specified in the ctor? Besides, passing it here means that this is not overriding the abstract `record_start` in `MetricBase`?

Reply:
Well python does not call the parent method if a subclass has the method name. 

```
class A:

    def __init__(self):
        self.name = "A"

    def get_name(self):
        return self.name

class B(A):

    def __init__(self):
        super().__init__()
        self.name = "B"

    def get_name(self, name):
        return name

if __name__ == "__main__":
    b = B()
    # print(b.get_name("garrett"))
    print(b.get_name())
```

Reply:
> can it be specified in the ctor

I think this is a better approach because it also deals with the issue of the metric recording on different devices. https://github.com/pytorch/pytorch/pull/57454#discussion_r625474043